repos/system_upgrade/common/actors/checkmountoptions/libraries/checkmountoptions.py

```diff
@@ -0,0 +1,77 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+
+def inhibit_upgrade_due_var_with_noexec(mountpoint, found_in_fstab=False):
+    summary = (
+        'Leapp detected that the {0} mountpoint is mounted with the "noexec" option, '
+        'which prevents binaries necessary for the upgrade from being executed. '
+        'The upgrade process cannot continue with {0} mounted using the "noexec" option.'
+    )
+
+    if found_in_fstab:
+        hint = (
+            'Temporarily remove the "noexec" option from {0} entry in /etc/fstab until the system is upgraded, '
+            'and remount the partition without the "noexec" option.'
+        )
+        related_resource = [reporting.RelatedResource('file', '/etc/fstab')]
+    else:
+        hint = 'Remount {0} without the noexec option.'
+        related_resource = []
+
+    reporting.create_report([
+        reporting.Title(
+            'The {0} mountpoint is mounted with the "noexec" option.'.format(mountpoint)
+        ),
+        reporting.Summary(summary.format(mountpoint)),
+        reporting.Remediation(hint=hint.format(mountpoint)),
+        reporting.RelatedResource('file', '/etc/fstab'),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Tags([reporting.Tags.FILESYSTEM]),
+        reporting.Flags([reporting.Flags.INHIBITOR]),
+    ] + related_resource)
+
+
+def find_mount_entry_with_mountpoint(mount_entries, mountpoint):
+    for mount_entry in mount_entries:
+        if mount_entry.mount == mountpoint:
+            return mount_entry
+    return None
+
+
+def find_fstab_entry_with_mountpoint(fstab_entries, mountpoint):
+    for fstab_entry in fstab_entries:
+        if fstab_entry.fs_file == mountpoint:
+            return fstab_entry
+    return None
+
+
+def check_noexec_on_var(storage_info):
+    """Check for /var or /var/lib being mounted with noexec mount option."""
+
+    var_lib_fstab_entry = find_fstab_entry_with_mountpoint(storage_info.fstab, '/var/lib')
+    if var_lib_fstab_entry:
+        if 'noexec' in var_lib_fstab_entry.fs_mntops.split(','):
+            inhibit_upgrade_due_var_with_noexec(var_lib_fstab_entry.fs_file, found_in_fstab=True)
+            return  # Do not check further as present mounts would likely reflect fstab
+    else:
+        var_fstab_entry = find_fstab_entry_with_mountpoint(storage_info.fstab, '/var')
+        if var_fstab_entry and 'noexec' in var_fstab_entry.fs_mntops.split(','):
+            inhibit_upgrade_due_var_with_noexec(var_fstab_entry.fs_file, found_in_fstab=True)
+            return  # Do not check further as present mounts would likely reflect fstab
```

Comment:
See my comment above. Regarding that, the solution here could look like this (pseudocode):
```python
    noexec_fstab_entries = []
    for mountpoint in ['/var', '/var/lib']:
        fstab_entry = find_fstab_entry_with_mountpoint(storage_info.fstab, '/var/lib')
        if fstab_entry and 'noexec' in fstab_entry.fs_mntops.split(','):
            noexec_fstab_entries.append(fstab_entry)
    if noexec_fstab_entries:
        inhibit_upgrade_due_var_with_noexec(noexec_fstab_entries, found_in_fstab=True)
        return  # Do not check further as present mounts would likely reflect fstab
```
similar for the block below.

Reply:
btw, in case it's implemented like that, `/var/lib/leapp` could be added as well for free.

Reply:
The code was written in a specific manner to account for mount option "shadowing" - for example:
- `/var` is mounted with `noexec`
- `/var/lib` is mounted *without* noexec
presents no problem for leapp (I've tried the described scenario), however, the suggested change would mark it as an inhibitor for the upgrade.

Reply:
you are right!! Important is just the last mountpoint on the path. Where I left my head? :-D Is it monday?